pkg/controller/daemon/controller.go

```diff
@@ -316,8 +339,19 @@ func (dsc *DaemonSetsController) updateNode(old, cur interface{}) {
 		// A periodic relist will send update events for all known pods.
 		return
 	}
+	dsList, err := dsc.dsStore.List()
+	if err != nil {
+		glog.Errorf("Error enqueueing daemon sets: %v", err)
+		return
+	}
+	for i := range dsList.Items {
+		ds := &dsList.Items[i]
+		shouldEnqueue := (checkDsNodeMatch(ds, oldNode) || checkDsNodeMatch(ds, curNode))
```

Comment:
Can we also exclude the case where the oldNode shouldEnqueqe and the curNode shouldEnqueue? Presumably, that would me "do nothing". i.e. make the OR an XOR